python/llm/test/inference_gpu/test_optimize_model.py

```diff
@@ -56,5 +60,98 @@ def test_optimize_model(Model, Tokenizer, model_path):
     assert any(diff) is False
 
 
+class Test_Optimize_Gpu_Model:
+    def setup(self):
+        self.lower_bound = 3e-3
+        
+        self.layer_inputs = []
+        self.layer_outputs = []
+
+    def run_optimize_gpu_model(self, Model, Tokenizer, model_path, self_attn):
+        def forward_hook(module, output, layer_name):
+            self.layer_outputs.append(output)
+
+        def pre_hook(module, input):
+            self.layer_inputs.append(input)
+
+        def load_pre_hook(module, input):
+            return self.layer_inputs[0]
+
+        tokenizer = Tokenizer.from_pretrained(model_path, trust_remote_code=True)
+        input_ids = tokenizer.encode(prompt, return_tensors="pt").to(device)
+
+        model = Model.from_pretrained(model_path,
+                                      load_in_4bit=True,
+                                      optimize_model=False,
+                                      trust_remote_code=True)
+        model = model.to(device)
+
+        for layer_name, layer_module in model.named_modules():
+            if layer_name == self_attn:
+                layer_module.register_forward_pre_hook(
+                    lambda module, input: pre_hook(module, input))
+                layer_module.register_forward_hook(
+                    lambda module, output, layer_name=layer_name: forward_hook(module, 
+                                                                               output, layer_name))
+        logits_base_model = (model(input_ids)).logits
+        # the list `layer_output` has only one element.
+        layer_tensor = self.layer_outputs.pop()
```

Comment:
use layer_outputs[0]

Reply:
this should be `pop` to clear the list, or there will be multi values in next forward.